pycloudlib/azure/cloud.py

```diff
@@ -633,6 +722,14 @@ def launch(
         name=None,
         inbound_ports=None,
         username: Optional[str] = None,
+        resource_group: Optional[util.AzureParams] = None,
+        network_security_group_: Optional[util.AzureCreateParams] = None,
+        virtual_network_: Optional[util.AzureCreateParams] = None,
+        subnet_: Optional[util.AzureCreateParams] = None,
+        ip_addresses: Optional[List[Optional[util.AzureCreateParams]]] = None,
+        network_interface_clients: Optional[
```

Comment:
I think it would be clearer to name this parameter as this, as it contains parameters to create a network interfaces, not clients.

```suggestion
        network_interfaces: Optional[
```